apps/sr-frontend/src/app/CharacterSelector.tsx

```diff
@@ -1,20 +1,33 @@
 import { CardThemed } from '@genshin-optimizer/common/ui'
+import type { CharacterKey } from '@genshin-optimizer/sr/consts'
 import {
   CharacterAutocomplete,
   useCharacterContext,
+  useDatabaseContext,
 } from '@genshin-optimizer/sr/ui'
 import { CardContent, Container } from '@mui/material'
+import { useCallback } from 'react'
 
 export default function CharacterSelector() {
   const { characterKey, setCharacterKey } = useCharacterContext()
+  const { database } = useDatabaseContext()
+
+  const setOrCreateCharacter = useCallback(
+    (charKey: CharacterKey | '') => {
+      if (!charKey) return
```

Comment:
Unable to deselect a character. should `setCharacterKey` on `""`
```suggestion
      if(charKey) database.chars.getOrCreate(charKey)
      setCharacterKey(charKey)
```